### ISSUE TITLE
Remove Pjax from core and make as a peer dependency

### ISSUE THREAD

**Issue by bamccaul-dev** [Pageworks/djinnjs#98]
We need to make Pjax be a peer dependency for Djinn. We don't always need that functionality, but having it be baked into and leveraging Djinn has its benefits.

**Comment by codewithkyle** [comment]
I'm going to place this task on hold and we can discuss it later. It seems like we are trying to solve a problem that has already been solved. If a developer doesn't need Pjax navigation they can [disable pjax in their config file](https://djinnjs.com/guides/configuration/#pjax). When Pjax is disabled Djinn will not lazy load the Pjax + Worker scripts.

Edit: as a side note I did spend ~2 hours trying to figure out the easiest way to split Pjax into a peer dependency but it ended up being way to much work. I would prefer we keep it integrated and developers can continue to set `pjax: false` in the config if they don't need a persistent application state.
